protocols/horizon/main.go

```diff
@@ -499,34 +501,57 @@ type Transaction struct {
-	MaxFee             int64               `json:"max_fee,string"`
-	OperationCount     int32               `json:"operation_count"`
-	EnvelopeXdr        string              `json:"envelope_xdr"`
-	ResultXdr          string              `json:"result_xdr"`
-	ResultMetaXdr      string              `json:"result_meta_xdr"`
-	FeeMetaXdr         string              `json:"fee_meta_xdr"`
-	MemoType           string              `json:"memo_type"`
-	MemoBytes          string              `json:"memo_bytes,omitempty"`
-	Memo               string              `json:"memo,omitempty"`
-	Signatures         []string            `json:"signatures"`
-	ValidAfter         string              `json:"valid_after,omitempty"`
-	ValidBefore        string              `json:"valid_before,omitempty"`
-	FeeBumpTransaction *FeeBumpTransaction `json:"fee_bump_transaction,omitempty"`
-	InnerTransaction   *InnerTransaction   `json:"inner_transaction,omitempty"`
+	ID                string    `json:"id"`
+	PT                string    `json:"paging_token"`
+	Successful        bool      `json:"successful"`
+	Hash              string    `json:"hash"`
+	Ledger            int32     `json:"ledger"`
+	LedgerCloseTime   time.Time `json:"created_at"`
+	Account           string    `json:"source_account"`
+	AccountMuxed      string    `json:"account_muxed,omitempty"`
+	AccountMuxedID    uint64    `json:"account_muxed_id,omitempty,string"`
+	AccountSequence   string    `json:"source_account_sequence"`
+	FeeAccount        string    `json:"fee_account"`
+	FeeAccountMuxed   string    `json:"fee_account_muxed,omitempty"`
+	FeeAccountMuxedID uint64    `json:"fee_account_muxed_id,omitempty,string"`
+	FeeCharged        int64     `json:"fee_charged,string"`
+	MaxFee            int64     `json:"max_fee,string"`
+	OperationCount    int32     `json:"operation_count"`
+	EnvelopeXdr       string    `json:"envelope_xdr"`
+	ResultXdr         string    `json:"result_xdr"`
+	ResultMetaXdr     string    `json:"result_meta_xdr"`
+	FeeMetaXdr        string    `json:"fee_meta_xdr"`
+	MemoType          string    `json:"memo_type"`
+	MemoBytes         string    `json:"memo_bytes,omitempty"`
+	Memo              string    `json:"memo,omitempty"`
+	Signatures        []string  `json:"signatures"`
+	// Action needed in release: horizon-v3.0.0: remove valid_(after|before)
+	ValidAfter         string                    `json:"valid_after,omitempty"`
+	ValidBefore        string                    `json:"valid_before,omitempty"`
+	Preconditions      *TransactionPreconditions `json:"preconditions,omitempty"`
+	FeeBumpTransaction *FeeBumpTransaction       `json:"fee_bump_transaction,omitempty"`
+	InnerTransaction   *InnerTransaction         `json:"inner_transaction,omitempty"`
+}
+
+type TransactionPreconditions struct {
+	Timebounds   *TransactionPreconditionsTimebounds   `json:"timebounds,omitempty"`
+	Ledgerbounds *TransactionPreconditionsLedgerbounds `json:"ledgerbounds,omitempty"`
```

Comment:
❓ Since we normalized on uppercase bounds in the sdk, should this be:
```suggestion
	TimeBounds   *TransactionPreconditionsTimeBounds   `json:"timebounds,omitempty"`
	LedgerBounds *TransactionPreconditionsLedgerBounds `json:"ledgerbounds,omitempty"`
```

Reply:
Good call; might as well!